tracer/src/Datadog.Trace/SpanLink.cs

```diff
@@ -25,39 +25,24 @@ internal class SpanLink
     /// in OpenTelemetry that's called a Span Context, which may also include tracestate and trace flags.
     /// </summary>
     /// <param name="spanLinkContext">The context of the spanlink to extract attributes from</param>
-    /// <param name="decoratedSpan">Reference to the span you're adding SpanLinks to</param>
     /// <param name="attributes">Optional dictionary of attributes to take for the spanlink.</param>
-    public SpanLink(SpanContext spanLinkContext, Span decoratedSpan, List<KeyValuePair<string, string>>? attributes = null)
+    public SpanLink(SpanContext spanLinkContext, List<KeyValuePair<string, string>>? attributes = null)
     {
         Context = spanLinkContext;
-        DecoratedSpan = decoratedSpan;
         Attributes = attributes;
     }
 
-    public SpanLink(Span spanToLink, Span decoratedSpan, List<KeyValuePair<string, string>>? attributes = null)
-        : this(spanToLink.Context, decoratedSpan, attributes)
-    {
-    }
-
     public List<KeyValuePair<string, string>>? Attributes { get; private set; }
 
     public SpanContext Context { get;  }
 
-    public Span DecoratedSpan { get; }
-
     /// <summary>
     /// Adds an Attribute to the SpanLink.
     /// </summary>
     /// <param name="name">name of attribute</param>
     /// <param name="value">value of attribute</param>
     public void AddAttribute(string name, string value)
     {
-        if (DecoratedSpan.IsFinished)
-        {
-            Log.Warning("AddAttribute should not be called after the decorated span was closed");
-            return;
-        }
-
```

Comment:
With this removed we technically now enable the ability to add attributes to a `SpanLink` after the `Span` is finished. Are we okay with that? I see that the tests asserting that prior behavior were removed. I think it is fine, just wanted to call out a potential behavior change

Reply:
I thought about suggesting the removal of `AddAttribute` but it appears to be requirement from the RFC for this API.

Reply:
Removing the `AddAttribute` API is a good idea. Since this is not a public API, I think that it should be fine to remove so that we match the .NET API, and if there's any issues we could change it back in the future